### PR TITLE
Clarify Scales hierarchy

### PR DESCRIPTION
**Pull request by bryevdv**
pre-work for #6517 

flying blind on JS tests as `node make test` won't run for me locally again, whether I start chrome headless myself or not.

### CODE REVIEW

**Comment by bryevdv** [comment]
@mattpap the js tests to be "failing fast" I can't run locally, I only see one error on CI, I fix it and then another shows up. This is not a productive way to work.

**Comment by mattpap** [comment]
> This is not a productive way to work.

^^^ Actually, this isn't a productive way to work. You could have started an issue describing the problem and helping me out resolving it, especially if this is platform dependent issue. I've noticed that there are build related issues myself yesterday, possibly due to changed dependencies. Though it may be a different issue altogether.

File: bokehjs/src/lib/models/scales/scale.ts [hunk 1]
```diff
@@ -50,4 +50,50 @@ export abstract class Scale extends Transform {
     else
       return [this.invert(sx0), this.invert(sx1)]
   }
+
+  // These are here on Scale because TypeScript does not support mixins. These
```
Comment (mattpap) [inline bokehjs/src/lib/models/scales/scale.ts]: It does support mixins, just differently from what one would do in Python. `Signalable` is a great example of such.

**Comment by bryevdv** [comment]
I'm frustrated, Mateusz, in part because there is not really any useful information to provide:
```

[08:39:36] Finished 'test:size' after 693 ms
[08:39:36] Starting 'test:start:headless'...
[08:39:36] Failed 'test:start:headless' after 4 ms
[08:39:36] Starting 'test:start:server'...
listening on 127.0.0.1:5777
[08:39:38] Finished 'test:start:server' after 1.73 s
[08:39:38] Starting 'defaults:generate'...
[08:39:39] generate_defaults.py: Wrote models_defaults.json with 179 model classes
[08:39:39] generate_defaults.py: Wrote widgets_defaults.json with 23 model classes
[08:39:39] Finished 'defaults:generate' after 1.27 s
[08:39:39] Starting 'test:compile'...
[08:39:43] Finished 'test:compile' after 3.97 s
[08:39:43] Starting 'test:unit:bundle'...
[08:39:48] Finished 'test:unit:bundle' after 5.07 s
[08:39:48] Starting 'test:integration:bundle'...
[08:39:49] Finished 'test:integration:bundle' after 44 ms
[08:39:49] failed: task 'test' failed
[08:39:49]     at _run (/Users/bryan/work/bokeh/bokehjs/make/task.ts:134:26)
[08:39:49]     at process._tickCallback (internal/process/next_tick.js:68:7)

~/work/bokeh/bokehjs master 18s
```

**Comment by mattpap** [comment]
> `Failed 'test:start:headless'`

That's all I need to know. It's interesting because out of the sudden I can reproduce this in one of my working environments, which is weird, because it fails only in one and `which` (run in a terminal) works. So there are two issues here. One is that the nodejs equivalent of `which` apparently doens't work, and the other is that intermediate errors are not showed, because you should have seen this:
```
Error: can't find chromium or chrome executables
```
Starting your own chrome doesn't help, because currently reusing a running instance is not supported. I will implement this as part of the overall fix.

**Comment by bryevdv** [comment]
@mattpap I am still unable to run tests locally after rebasing on current master, either with or without chrome headless manually started. Do you have any suggestions or workarounds?

**Comment by bryevdv** [comment]
Edit actually it does run, I had copied a start command out of my history that had a different port or some reason. I can see the test failures locally now.

**Pull request merged**